numpy-stubs/__init__.pyi

```diff
@@ -29,11 +30,17 @@ if sys.version_info[0] < 3:
 else:
     from typing import SupportsBytes
 
+# Used to type hint values that could be in `ndarray`
+_T = TypeVar("_T")
+
 _Shape = Tuple[int, ...]
 
 # Anything that can be coerced to a shape tuple
 _ShapeLike = Union[int, Sequence[int]]
 
+# Anything that mentioned as `array_like ` in  documentation
+_ArrayLike = Union[_T, Sequence[_T]]
```

Comment:
I support the idea of clearly document NumPy's interface, but in practice, how does this specific annotation differ from `Any`?

I get the idea of trying to constrain to only "array-like" inputs, but keep in mind that for NumPy that means practically anything -- we can coerce almost any type of object to ndarray with `np.array()`.

Reply:
I was trying to prepare `ndarray` to become generic and limit passed argument to have same `dtype` as `self` but after a bit more testing I agree with you. It's better to put `Any` here.